test/controller/mixins/spec.back-links.js

```diff
@@ -78,153 +65,93 @@ describe('mixins/back-links', () => {
+            controller.options.route = '/step3';
             controller.backlinksSetLocals(req, res, next);
             res.locals.backLink.should.equal('/base/step2');
         });
 
         it('returns undefined if referrer header is not on whitelist', () => {
             req.get.withArgs('referrer').returns('http://example.com/not-whitelisted');
             controller.options.route = '/step3';
-            req.sessionModel.set('steps', ['/step1', '/step2']);
             controller.options.backLinks = ['whitelist'];
             controller.backlinksSetLocals(req, res, next);
             expect(res.locals.backLink).to.be.undefined;
         });
 
         it('returns undefined if referrer header not present', () => {
             controller.options.route = '/step3';
-            req.sessionModel.set('steps', ['/step1', '/step2']);
             controller.options.backLinks = ['whitelist'];
             controller.backlinksSetLocals(req, res, next);
             expect(res.locals.backLink).to.be.undefined;
         });
 
-        describe('isBackLink request object property', () => {
-
-            it('is false when the last step is before the current page', () => {
-                req.sessionModel.set('steps', ['/step1', '/step2']);
-                controller.options.route = '/step3';
-                controller.backlinksSetLocals(req, res, next);
-                req.isBackLink.should.not.be.ok;
-                res.locals.isBackLink.should.not.be.ok;
-            });
-
-            it('is true when the last step matches the page path', () => {
-                req.sessionModel.set('steps', ['/step1', '/step2']);
-                controller.options.route = '/step2';
-                controller.backlinksSetLocals(req, res, next);
-                req.isBackLink.should.be.ok;
-                res.locals.isBackLink.should.be.ok;
-            });
-
-            it('is true when the last step matches the next path', () => {
-                req.sessionModel.set('steps', ['/step1', '/step2']);
-                controller.options.next = '/step2';
-                controller.options.route = '/step1';
-                controller.backlinksSetLocals(req, res, next);
-                req.isBackLink.should.be.ok;
-                res.locals.isBackLink.should.be.ok;
-            });
-
+        it('skips history items that are marked as skip', () => {
+            req.journeyModel.set('history', [
+                { path: '/base/step1', next: '/base/step2' },
+                { path: '/base/step2', next: '/base/step3' },
+                { path: '/base/step3', next: '/base/step4', skip: true },
+                { path: '/base/step4', next: '/base/step5', skip: true },
+                { path: '/base/step5', next: '/base/step6' }
+            ]);
+            controller.options.route = '/step5';
+            controller.backlinksSetLocals(req, res, next);
+            res.locals.backLink.should.equal('/base/step2');
         });
+
```

Comment:
Appreciate the functionality is covered in backlinksSetLocals tests, but is getBackLink intended to be implementable, and should it have its own tests?